sdcm/sct_config.py

```diff
@@ -2115,34 +2477,36 @@ def _dotted_get(self, key: str):
             current = current.get(k)
         return current
 
-    def _validate_value(self, opt):
-        opt['is_k8s_multitenant_value'] = False
+    def _validate_value(self, field: Field):
+        field.json_schema_extra['is_k8s_multitenant_value'] = False
+
+        def no_op(x):
+            return x
+
+        for annotation in field.metadata:
+            if isinstance(annotation, BeforeValidator):
+                from_env_func = annotation.func
+                break
+        else:
+            from_env_func = no_op
```

Comment:
this repeats, could be moved to function `get_field_cast_function` or smthing like this.